c7n/actions/network.py

```diff
@@ -137,6 +161,22 @@ def get_groups_by_names(self, names):
                 names=list(unresolved), groups=[g['GroupId'] for g in sgs]))
         return sgs
 
+    def get_groups_by_tag(self, key, values, vpc_id):
+        """Get security groups that match tag values."""
```

Comment:
do we already have logic to detect over 5 total sgs? else this could result in a errors.

Reply:
@kapilt I added the raise statement handle it.